project: allow loading from user-specified filename

cmatsuoka opened the pull request:
Allow loading the spread project from a filename specified by the
user using an environment variable. This was chosen over a command
line option to avoid encouraging projects to use arbitrary random
filenames.

If the environment variable is not set, current behavior is kept
and the project will be loaded from `spread.yaml` or `.spread.yaml`
in the current or parent directories. If the variable is set, parent
directories will not be searched.

This is an experimental, undocumented feature and may be removed or
modified without prior notice.

Signed-off-by: Claudio Matsuoka <cmatsuoka@gmail.com>

cmatsuoka committed:
project: allow loading from user-specified filename

Allow loading the spread project from a filename specified by the
user using an environment variable. This was chosen over a command
line option to avoid encouraging projects to use arbitrary random
filenames.

If the environment variable is not set, current behavior is kept
and the project will be loaded from `spread.yaml` or `.spread.yaml`
in the current or parent directories. If the variable is set, parent
directories will not be searched.

This is an experimental, undocumented feature and may be removed or
modified without prior notice.

Signed-off-by: Claudio Matsuoka <cmatsuoka@gmail.com>

lengau commented on spread/project_test.go:
```diff
@@ -78,19 +77,42 @@ suites:
+		{"spread.yaml", "", ""},
+		{".spread.yaml", "", ""},
+		{"other.yaml", "", "cannot load project from .*: cannot find spread.yaml or .spread.yaml"},
+		{"custom.yaml", "custom.yaml", ""},
```
While it doesn't really affect this implementation, it might be a good idea to have a test that makes the file in a subdirectory so that if the implementation of `Load()` changes we don't accidentally drop that use case.

cmatsuoka replied:
This was a necessary test because loading from a subdir was actually broken. When no custom filename is specified, the project path is where `spread.yaml` is located, and can be in a parent directory. But if we specify a filename, potentially with a relative or absolute path, we'll keep the project path in the current directory without verifying parents.

cmatsuoka committed:
fix: allow loading project from a subdir

If a user-specified project file name is used, don't change the
project path based on the project file location. Otherwise the
project path is where spread.yaml is located (can be a parent dir).

Signed-off-by: Claudio Matsuoka <cmatsuoka@gmail.com>

cmatsuoka committed:
refactor: better way to set project path

Signed-off-by: Claudio Matsuoka <cmatsuoka@gmail.com>
